### PR TITLE
Add some notes for potential shim reviews

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd-efi/issues/105

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -3,6 +3,76 @@
...
+
+The most important takeaway from this is that `fwupd.efi` does not *execute* the capsule itself,
+but instead calls the firmware capsule loader runtime interface which tells the firmware to copy
+the structured data into some part of the protected memory that will survive a reboot.
+It then reboots the machine, causing the machine to re-initialize the firmware, which in turn
+verifies the signature of the capsule payload and flashes the contents to NVRAM.
+The machine then reboots again, re-initializing the firmware and booting to the OS.
+
+For all vendors correctly implementing UEFI capsule updates, it is not possible to flash unsigned
+apsules, or capsules signed by the wrong platform key.
```
Comment (gmbr3) [inline SECURITY.md:70]: *capsules

**Review by gmbr3 — CHANGES_REQUESTED** [review]

File: SECURITY.md [hunk 2]
```diff
@@ -3,6 +3,76 @@
 Due to the nature of what we are doing, fwupd takes security very seriously.
 If you have any concerns please let us know.
 
+## Considerations For Secure Boot
+
+In most secure boot configurations `fwup*.efi` will be signed and run by a first-stage bootloade
```
Comment (superm1) [inline SECURITY.md:8]: bootloader

File: SECURITY.md [hunk 3]
```diff
@@ -3,6 +3,76 @@
...
+(e.g. Shim) and it is thus critically important that fwup-efi is not allowed to run unauthenticated
+or untrusted code.
+
+The following sections contain considerations about what the EFI binary actually does, which
+hopefully should give enough assurance that using `fwupd.efi` is thus no threat to the chain of
+trust required by secure boot.
+
+### General Logic Flow
+
```
Comment (superm1) [inline SECURITY.md:17]: how about putting a mermaid chart here?

File: SECURITY.md [hunk 4]
```diff
@@ -3,6 +3,76 @@
...
+* Find and validate update state variables:
+
+* Read the list of EFI variables, and match the ones with the fwupd GUID.
```
Comment (superm1) [inline SECURITY.md:20]: It's probably worth either linking to the fwupd GUID in code or putting it in parenthesis

File: SECURITY.md [hunk 5]
```diff
@@ -3,6 +3,76 @@
...
+* For each matching variable, parse this data as a `FWUP_UPDATE_INFO` structure, being mindful the
```
Comment (superm1) [inline SECURITY.md:21]: link to the code for this one

File: SECURITY.md (hunk 1 above)
Comment (hughsie) [reply SECURITY.md:70]: Fixed, thanks.

File: SECURITY.md (hunk 2 above)
Comment (hughsie) [reply SECURITY.md:8]: Fixed, thanks.

File: SECURITY.md (hunk 4 above)
Comment (hughsie) [reply SECURITY.md:20]: Good idea, fixed.

File: SECURITY.md (hunk 5 above)
Comment (hughsie) [reply SECURITY.md:21]: Fixed.

File: SECURITY.md (hunk 3 above)
Comment (hughsie) [reply SECURITY.md:17]: I've added something super simple.

**Review by superm1 — APPROVED** [review]

**Pull request merged**